frame/contracts-registry/src/types.rs

```diff
@@ -0,0 +1,28 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{sp_runtime::RuntimeDebug, traits::Get, BoundedVec};
+use scale_info::TypeInfo;
+
+/// A proposal index.
+pub type ContractIndex = u32;
+
+/// An active voter.
+#[derive(Clone, Encode, Decode, MaxEncodedLen, RuntimeDebug, TypeInfo)]
+#[codec(mel_bound(skip_type_params(MaxContractsLen, MaxParties)))]
+#[scale_info(skip_type_params(MaxContractsLen, MaxParties))]
+pub struct ContractsData<MaxContractsLen: Get<u32>, MaxParties: Get<u32>, AccountId> {
+	pub data: BoundedVec<u8, MaxContractsLen>,
+	pub parties: BoundedVec<AccountId, MaxParties>,
+	pub creator: AccountId,
+}
+
+impl<MaxContractsLen: Get<u32>, MaxParties: Get<u32>, AccountId>
+	ContractsData<MaxContractsLen, MaxParties, AccountId>
+{
+	pub fn new(
+		data: BoundedVec<u8, MaxContractsLen>,
+		parties: BoundedVec<AccountId, MaxParties>,
+		creator: AccountId,
+	) -> Self {
+		Self { data, creator, parties }
+	}
+}
```

Comment:
Doesn't look like this constructor is worth it. Why not just create ContractsData directly in `create_contract`?